viewing_party/party.py

```diff
@@ -1,23 +1,223 @@
 # -----------------------------------------
 
+def get_unique_watched(user_data):
+    # Get the list of movies the user watched
+    user_watched = user_data["watched"]
+    
+    # Get movies watched by friends
+    friends_watched = []
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friends_watched.append(movie)  
+
+    # Find movies the user watched but none of friends watched
+    user_unique_watched = []
+    for movie in user_watched:
+        if movie not in friends_watched: 
+            user_unique_watched.append(movie)
+
+    return user_unique_watched
+    
+def get_friends_unique_watched(user_data):
+    # Get the list of movies the user watched
+    user_watched = user_data["watched"]
+    
+    # Get movies watched by friends
+    friends_watched = []
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friends_watched.append(movie)  
+
+    # Find movies the user watched but none of friends watched
+    friends_unique_watched = []
+    for movie in friends_watched:
+        if movie not in user_watched and movie not in friends_unique_watched:
+            friends_unique_watched.append(movie)
+
+    return friends_unique_watched    
         
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    recommended_movies = []
+    user_watched_titles = []
+    unique_recommendations = []
+
+    # store titles of movies user has watched
+    for movie in user_data["watched"]:
+        user_watched_titles.append(movie["title"])
+
+    # check each friend's watched list
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            # make sure user hasn't watched it and it's on a subscribed service
+            if (
+                movie["title"] not in user_watched_titles
+                and movie["host"] in user_data["subscriptions"]
+                and movie["title"] not in unique_recommendations  # avoid duplicates
+            ):
+                recommended_movies.append(movie)
+                unique_recommendations.append(movie["title"])  # track added movies
+
+    return recommended_movies
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+    # Step 1: Count genres
+    genre_count = {}
+    for movie in user_data["watched"]:
+        genre = movie["genre"]
+        if genre in genre_count:
+            genre_count[genre] += 1
+        else:
+            genre_count[genre] = 1
+
+    # Step 2: Find the most watched genre    
+    most_watched_genre = None
+    highest_count = 0
+    
+    for genre, count in genre_count.items():
+        if count > highest_count:
+            highest_count = count
+            most_watched_genre = genre

```

Comment:
Invoke the function you already wrote to find the most watched genre in order to DRY up your codebase
```suggestion
most_watched_genre = get_most_watched_genre(user_data)
```